### DISCUSSION TITLE
Code Update

### DISCUSSION THREAD

**Discussion by Lukylic** [tfyoung/esphome-junctek_kgf#14] [General]
Hello,
I did update your Code to add all avaliable sensors from Junctek. (28)
do you want me to send you this new code?
if yess, how I can send it to you?

**Comment by tronron** [comment]
I for one, would love to have a code update to read all the sensors!

**Comment by Lukylic** [comment]
[junctek_luckylic.zip](https://github.com/tfyoung/esphome-junctek_kgf/files/11299064/junctek_luckylic.zip)
I do not know if you can loald it by this way, you can try. This update is probably not perfect but is running

**Reply by tfyoung** [reply]
Thanks for doing this work, I haven't had any time to dedicate to this lately, but I will try and take a look when I get a chance!

**Reply by Lukylic** [reply]
Great! Thank for the community.

**Reply by jojora** [reply]
hello
What is the version of esphome with which it can be installed?

**Reply by alanv72** [reply]
> [junctek_luckylic.zip](https://github.com/tfyoung/esphome-junctek_kgf/files/11299064/junctek_luckylic.zip) I do not know if you can loald it by this way, you can try. This update is probably not perfect but is running

forked with your updated code with edits for latest firm of a KG160F.
https://github.com/alanv72/esphome-junctek_kgf

**Comment by Catchmydrift** [comment]
> Hello, I did update your Code to add all avaliable sensors from Junctek. (28) do you want me to send you this new code? if yess, how I can send it to you?

Hi, i would be very greatful if you could send it to me? Or post it up here to this reply?  Thank you 👍
Ucatchmydrift@protonmail.com

**Comment by dann2008** [comment]
I also woud like to see the POWER sensor, its very useful to see the instant power charging or discahrging

**Comment by Kicosta** [comment]
> Hello, I did update your Code to add all avaliable sensors from Junctek. (28) do you want me to send you this new code? if yess, how I can send it to you?


Hi Lukylic, I have two KG140Fs that I connected via RS485 using two different addresses and a screen connected to the first one and so I can see the values ​​of both by changing the address in the screen settings. 
My question is if you have ever tested the possibility of connecting an esp32 with the RS485 interface in parallel with the RS485 that connects both devices so that you can read the values ​​of both by entering the address of each of the KGs in the yaml file .
In the tfyoung Yaml file the address is specified so I guess this can be done but I saw that your Yaml configuration does not have this feature. Thanks for any support and I hope I was clear.

Costa
